Add weyland-yutani-theme package.

jstaursky opened the pull request:
### Brief summary of what the package does

Emacs theme based off of the Alien movie franchise!

### Direct link to the package repository

https://github.com/jstaursky/weyland-yutani-theme

### Your association with the package

I am creator + maintainer

### Relevant communications with the upstream package maintainer

**None needed**

### Checklist

Please confirm with `x`:

- [X] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses).
- [X] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [X] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [X] My elisp byte-compiles cleanly
- [X] `M-x checkdoc` is happy with my docstrings
- [X] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [ ] I have confirmed some of these without doing them

riscy commented:
Nice theme. :)

### weyland-yutani-theme.el ###

byte-compile (using Emacs 26.3):
- weyland-yutani-theme.el#L401: Don't set `no-byte-compile: t`.
  Byte-compiling is enabled in what follows
- Per CONTRIBUTING.org, consider using lexical binding.
  See: https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org#fixing-typical-problems
- Lexical binding will be used in what follows
- No issues!

(I'm curious, where did you get the no-byte-compile: t from?  I see this in quite a few themes.)

checkdoc (using version 0.6.1):
```
Warning (emacs): 
weyland-yutani-theme.el:39: Argument `L' should appear (as L) in the doc string
```
Minor but worth doing.  Variables are (usually) downcased in elisp, as well.

package-lint (using version 20200419.406):
```
12 issues found:
1:0: error: Package should have a Homepage or URL header.
1:0: warning: The package summary should not end with a period.
1:0: error: There is no (provide 'weyland-yutani-theme) form.
30:0: error: Package should have a non-empty ;;; Commentary section.
34:10: error: You should depend on (emacs "24.1") if you need `color'.
100:6: warning: Closing parens should not be wrapped onto new lines.
109:28: error: You should depend on (emacs "24.1") if you need `color-lighten-name'.
195:6: error: You should depend on (emacs "24.1") if you need `org-date'.
199:6: error: You should depend on (emacs "24.1") if you need `org-block'.
388:4: warning: Closing parens should not be wrapped onto new lines.
389:3: warning: Closing parens should not be wrapped onto new lines.
391:2: warning: Closing parens should not be wrapped onto new lines.
```
It turns out the third is not necessary since `provide-theme` makes a call to `provide` -- but you can add the `provide` anyway to pacify package-lint.  Most of the nags about emacs version are extraneous, _except_ that you'll definitely want Emacs 24.1 for `color-lighten-name`.

Suggestions/experimental static checks:
- weyland-yutani-theme.el#L26: Prefer `https` over `http` if possible ([why?](https://news.ycombinator.com/item?id=22933774))

jstaursky commented:
Thanks for the compliment, a lot work went into it. Not sure where the no-byte-compile idiom comes from, I just figured it was standard and kept it after the initial generation from emacs theme creator. I removed the dependency on the 'color package, added lexical-binding in the header and all errors + warnings from package-lint should now be fixed. :)

riscy commented:
Here's the latest (looks like some inadvertent byte-compile warnings):

### weyland-yutani-theme.el ###

byte-compile (using Emacs 26.3):
```
weyland-yutani-theme.el:38:1:Warning: Unused lexical variable `quote'
weyland-yutani-theme.el:52:1:Warning: Unused lexical variable `Gold-alt'
weyland-yutani-theme.el:52:1:Warning: Unused lexical variable `darkPurple'
weyland-yutani-theme.el:52:1:Warning: Unused lexical variable
    `lightarcticblue'
weyland-yutani-theme.el:52:1:Warning: Unused lexical variable
    `bg-CharlesGreen'
weyland-yutani-theme.el:52:1:Warning: Unused lexical variable `bg-ArcticBlue'
weyland-yutani-theme.el:52:1:Warning: Unused lexical variable `bg-light-Black'
```

checkdoc (using version 0.6.1):
```
Warning (emacs): 
weyland-yutani-theme.el:39: Argument `l' should appear (as L) in the doc string
Warning (emacs): 
weyland-yutani-theme.el:433: White space found at end of line
```

package-lint (using version 20200419.406):
- No issues!

jstaursky commented:
Fixed checkdoc but insofar as byte-compile, I'm confused on the 'quote' variable its referring to - I don't see it anywhere. The other variables I plan on keeping as they may prove useful if/when further modes are to be supported.

riscy commented:
`'` is syntactic sugar for the `quote` function, so I suspect you have some unintentional syntax within `weyland-yutani-theme-face-specifier` if that helps.

Regarding the others, you can ensure everything byte-compiles cleanly (silently) by prefixing the variable names with `_`, e.g. `_dark-purple`, if you need to use them as placeholders.  You should use `hyphenated-variable-names` instead of `camelCase`, too -- it's just the community for Emacs lisp.

Here's the latest for reference.

### weyland-yutani-theme.el ###

byte-compile (using Emacs 26.3):
```
weyland-yutani-theme.el:38:1:Warning: Unused lexical variable `quote'
weyland-yutani-theme.el:52:1:Warning: Unused lexical variable `Gold-alt'
weyland-yutani-theme.el:52:1:Warning: Unused lexical variable `darkPurple'
weyland-yutani-theme.el:52:1:Warning: Unused lexical variable
    `lightarcticblue'
weyland-yutani-theme.el:52:1:Warning: Unused lexical variable `bg-ArcticBlue'
weyland-yutani-theme.el:52:1:Warning: Unused lexical variable `bg-light-Black'
```

checkdoc (using version 0.6.1):
```
Warning (emacs): 
weyland-yutani-theme.el:187: White space found at end of line
```

riscy commented:
Ah wait, I see it:

```elisp
  (let (res '())
```

Should be

or
```elisp
  (let ((res nil)) ...
```
or simply
```elisp
  (let (res) ...
```

jstaursky commented:
Ok
- fixed the byte-compile warnings (thanks for helping find the quote error).
  - reserved/placeholder variables are now prefixed with`_`
- Think I fixed check-doc (I'm not seeing the error your seeing--but tried to fix it anyway).

Now the reason I departed from the Emacs lisp community style is because I use the rainbow-colors package. Rainbow-colors causes certain variable names to be unintentionally highlighted (e.g. a variable named `white = "#C3D0DF"` will show up in bright white on the LHS and as the color I intend on the RHS of the equals sign) this became very distracting. So I used Capitals for ColorNames appearing in variable names.

However, there were some inconsistencies with variable name prefixes (e.g. `light-Blue` and `darkBlue`) and so I tried to meet you half way by making variable names be written as `<adjective>-<ColorName>`

riscy commented:
Thanks for all that.  Welcome to MELPA!
